lib/myxql.ex

```diff
@@ -122,6 +123,19 @@ defmodule MyXQL do
       iex> {:ok, pid} = MyXQL.start_link(after_connect: &MyXQL.query!(&1, "SET time_zone = '+00:00'"))
       {:ok, #PID<0.69.0>}
 
+  ## SSL
+
+  When using `ssl: true`, it's recommended to use non-empty `:ssl_opts` given the defaults are
+  insecure.
+
+  Make sure to enable public key, hostname, and other verification mechanisms appropriate for your
+  MySQL server installation, by passing options like `:verify_fun` which allows you to write
+  your own verification function. You may also consider using packages like
+  [`ssl_verify_fun`](https://hex.pm/packages/ssl_verify_fun).
+
+  See [`:ssl.tls_client_option`](http://erlang.org/doc/man/ssl.html#type-tls_client_option) in
+  ssl documentation for a list of all available options.
```

Comment:
This is a bit thin on the details. SSL is usually complex and telling people to read the docs is not going to help much. Can we follow up a list of guidelines, similar to how @voltone wrote?

I.e.:

  1. Use OTP 21
  2. If using a private ssl, do this
  3. if accessing a cloud ssl, do this

?

Reply:
I think most of the needed information is included in my comments in #61, though it may have to be rephrased a bit for use in formal docs. I'd be happy to review the result.

BTW, it may also be worth pointing out that using TLS without server certificate verification is a legitimate use-case: it does not protect against active attackers, but it does ensure information is encrypted in-flight, safe from passive observers. So depending on your threat model, non-verifying TLS may be better than no TLS at all.

If that's understood by the user, the 'empty :ssl_opts' warning can be suppressed by setting `ssl_opts: [verify: :verify_none]`, which I think properly captures what's being done.